### ISSUE TITLE
what are log_theta and division by dim doing

### ISSUE THREAD

**Issue by Isa-rentacs** [xcyao00/BGAD#13]
Hello, I have a question regarding these two lines during the ML loss:
https://github.com/xcyao00/BGAD/blob/main/engines/bgad_fas_train_engine.py#L72-L73

Could you elaborate the reason you are dividing the loss by dim, and apply `log_theta`? I'm trying to locate it in your paper, but couldn't find it.

**Comment by xcyao00** [comment]
Dividing the loss by dim is commonly used in normalizing flow, as the loss is calculated by summing the channel dimension. The log_theta is torch.nn.LogSigmoid, you can see the loss figure in pytorch docs, this is to make the loss more smooth around 0. When logp is negative, the log_theta is almost linear. At the beginning, the logp may be greater than 0, the log_theta can also ensure that there are still gradients.

**Comment by Isa-rentacs** [comment]
Understood, thank you for your clarification!
